lib/Response.php

```diff
@@ -153,8 +153,8 @@ public function setStatus($status): void
                 $statusCode,
                 $statusText
             ) = explode(' ', $status, 2);
-            $statusCode = (int) $statusCode;
         }
+        $statusCode = (int) $statusCode;
         if ($statusCode < 100 || $statusCode > 999) {
```

Comment:
in case this condition holds for all paths, you could restrict `$this->status` property to `int<100, 999>`

Reply:
That is true - we only ever store a 3-digit int in that variable. I pushed a commit.